cmd/sys-mgmt-executor/main.go

```diff
@@ -148,19 +166,99 @@ func runDockerCommands(service string, operation string) error {
 			expectedStatus = true
 		case STOP:
 			expectedStatus = false
+		case METRICS:
+
 		default:
-			// impossible to get here
 			panic(fmt.Sprintf("invalid operation %s", operation))
 		}
 
 		correctStatus, err := checkDockerContainerStatus(service, expectedStatus)
 		if err != nil {
-			return err
+			return "", err
 		}
 		if !correctStatus {
-			return fmt.Errorf("docker %s operation failed for service %s", operation, service)
+			return "", fmt.Errorf("docker %s operation failed for service %s", operation, service)
+		}
+		return "", nil
+	} else if operation == METRICS {
+
+		LoggingClient.Info(fmt.Sprintf("run the docker stats on service (%s) for operation (%s)", service, operation))
+		cmd := exec.Command("docker", "stats")
+		LoggingClient.Info(fmt.Sprintf("finished running the docker stats command"))
+
+		var stdout, stderr []byte
+		var errStdout, errStderr error
+		stdoutIn, _ := cmd.StdoutPipe()
+		stderrIn, _ := cmd.StderrPipe()
+		err := cmd.Start()
+		if err != nil {
+			log.Fatalf("cmd.Start() failed with '%s'\n", err)
+		}
+
+		// Note that cmd.Wait() should be called only AFTER we finish reading
+		// from stdoutIn and stderrIn.
+		// The wait group (wg) ensures that we finish
+		var wg sync.WaitGroup
+		wg.Add(1)
+		ticker := time.NewTicker(time.Second)
+		go func(ticker *time.Ticker) {
+			stdout, errStdout = copyAndCapture(os.Stdout, stdoutIn)
+			wg.Done()
+			now := time.Now()
+			for range ticker.C {
+				LoggingClient.Info(fmt.Sprintf("%s", time.Since(now)))
+			}
+		}(ticker)
+
+		// Create a timer that will kill the process
+		timer := time.NewTimer(time.Second * 3)
+		go func(timer *time.Timer, ticker *time.Ticker, cmd *exec.Cmd) {
+			for range timer.C {
+				cmd.Process.Signal(os.Kill)
+				ticker.Stop()
+			}
+		}(timer, ticker, cmd)
+		stderr, errStderr = copyAndCapture(os.Stderr, stderrIn)
+
+		wg.Wait()
+
+		err = cmd.Wait()
+		if err != nil {
+			LoggingClient.Error(fmt.Sprintf("cmd.Run() failed with %s\n", err.Error()))
+			log.Fatalf("cmd.Run() failed with %s\n", err)
```

Comment:
Why is `log.Fatal` used here? Are we unable to recover gracefully from the error which has been logged already through the LoggingClient?